### PR TITLE
add generic Mul operator

### PR DESCRIPTION
**Pull request by okdshin**
Add `Mul` operator to generic context.
It is very naive implementation. It doesn't support broadcasting.

### CODE REVIEW

**Comment by okdshin** [comment]
I failed to modify codes to merge master. I'll push force and retry.

**Comment by okdshin** [comment]
Could you check again? @msakai

File: menoh/mkldnn_with_generic_fallback/backend/generic/operator/mul.hpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        namespace generic_backend {
+            inline procedure make_mul(node const&,
+                                      std::vector<array> const& input_list,
+                                      std::vector<array> const& output_list) {
+                assert(input_list.size() == 2);
+                assert(output_list.size() == 1);
+
+                for(auto const& input : input_list) {
+                    if(input.dtype() != dtype_t::float_) {
+                        throw std::runtime_error("invalid dtype");
```
Comment (msakai) [inline menoh/mkldnn_with_generic_fallback/backend/generic/operator/mul.hpp]: Isn't it better to throw `menoh_impl::exception` with error code `menoh_error_code_invalid_dtype`?

File: menoh/mkldnn_with_generic_fallback/backend/generic/operator/mul.hpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                    }
+                }
+                if(total_size(input_list.at(0)) != total_size(input_list.at(1))) {
+                    throw std::runtime_error("broadcast is not supported yet");
```
Comment (msakai) [inline menoh/mkldnn_with_generic_fallback/backend/generic/operator/mul.hpp]: Similarly, throwing `menoh_impl::exception` with error code `menoh_error_code_dimension_mismatch` is more friendly?

File: menoh/mkldnn_with_generic_fallback/backend/generic/operator/mul.hpp (hunk 1 above)
Comment (okdshin) [reply menoh/mkldnn_with_generic_fallback/backend/generic/operator/mul.hpp]: I'll fix it.

File: menoh/mkldnn_with_generic_fallback/backend/generic/operator/mul.hpp (hunk 2 above)
Comment (okdshin) [reply menoh/mkldnn_with_generic_fallback/backend/generic/operator/mul.hpp]: I'll fix it.

**Comment by okdshin** [comment]
Thanks for review!

**Pull request merged**